### PR TITLE
15_2_6 Dynamic right border

### PR DESCRIPTION
**Pull request by sergeroshe**
@sergeroshe, description?

image

### CODE REVIEW

File: 15_mini_project_work/2_num_guessing_game.py [hunk 1]
```diff
@@ -29,22 +24,25 @@ def min_guaranteed_guess_count(left_border, right_border):
     return division_count
 
 
-def is_valid(input_string):
-    result = input_string.isdigit() and VALID_LEFT_BORDER <= int(input_string) <= VALID_RIGHT_BORDER
+def is_valid(input_string, guess_right_border):
+    result = input_string.isdigit() and guess_right_border <= int(input_string) <= guess_right_border
```
Comment (YanumVlad) [inline 15_mini_project_work/2_num_guessing_game.py:28]: Масло масляное - нужно зафиксать :)

File: 15_mini_project_work/2_num_guessing_game.py [hunk 2]
```diff
@@ -29,22 +24,25 @@ def min_guaranteed_guess_count(left_border, right_border):
...
     guess_count = 1
 
     while is_guess_wrong:
-        guess = input(PROMPT_MESSAGE)
-        if not is_valid(guess):
-            print(ERROR_MESSAGE)
+        guess_right_border = int(input(RIGHT_BORDER_PROMPT_MESSAGE))
+        hidden_num = random.randint(GUESS_LEFT_BORDER, guess_right_border)
+        guess = input(PROMPT_MESSAGE + str(guess_right_border) + COLON_SEP + '\n')
+        error_message = f'А может быть все-таки введем целое число от ' \
```
Comment (YanumVlad) [inline 15_mini_project_work/2_num_guessing_game.py:42]: Regression

**Review by YanumVlad — CHANGES_REQUESTED** [review]

File: 15_mini_project_work/2_num_guessing_game.py (hunk 1 above)
Comment (sergeroshe) [reply 15_mini_project_work/2_num_guessing_game.py]: Fixed.

File: 15_mini_project_work/2_num_guessing_game.py (hunk 2 above)
Comment (sergeroshe) [reply 15_mini_project_work/2_num_guessing_game.py]: Fixed.

File: 15_mini_project_work/2_num_guessing_game.py [hunk 3]
```diff
@@ -1,20 +1,17 @@
 import random
```
Comment (YanumVlad) [inline 15_mini_project_work/2_num_guessing_game.py:1]: Import waht you need only

**Review by YanumVlad — CHANGES_REQUESTED** [review]

**Review by YanumVlad — APPROVED** [review]

**Pull request merged**